src/tracer.ts

```diff
@@ -0,0 +1,97 @@
+export interface Tracer {
+  startSpan(id: string, startTime?: number): SpanAdapter;
+  decode(headers: Record<string, string>): SpanContext;
+}
+
+export interface SpanContext {
+  startSpan(id: string, startTime?: number): SpanAdapter;
+  encode(): Record<string, string>;
+}
+
+export interface SpanAdapter {
+  setAttribute(key: string, value: any): void;
+  context(): SpanContext;
+  end(): void;
```

Comment:
Can we add endTime as an optional argument? And since we are manually setting startTime it might be a good idea to manually set this from the clock too